test/config/fixtures/setup-file-no-isolate/setup.ts

```diff
@@ -0,0 +1,9 @@
+import { beforeEach } from 'vitest';
+
+export type MyContext = {
+  testOk: boolean
+}
+
+beforeEach<MyContext>((ctx) => {
+  ctx.testOk = true;
+});
```

Comment:
This doesn't really verify the fix. In #6682 the `afterEach/beforeEach` are not run for every test. It seemed that they are used like `after/before` instead.

Could we use a counter here and verifty that hooks are called for every single test? Something like:


```ts
const calls = { beforeEach: 0, afterEach: 0 };

beforeEach(() => {
  calls.beforeEach++;
})

afterEach(() => {
  calls.afterEach++;
})

// Verify that hooks were called for every test when isolation is disabled
after(() => {
  assert(calls.beforeEach === 2);
  assert(calls.afterEach === 2);
});
```

Reply:
Oh actually this test is enough. In my first reproduction I used global variables instead of test context to debug this. Those of course leaked from test to test when isolation was disabled.

Using test context seems to work: https://stackblitz.com/~/edit/vitest-dev-vitest-zwhfty?file=test/basic.test.ts&file=test/suite.test.ts:L3-L4&file=setup.ts:L2&initialPath=/__vitest__/